koap/apiTests/decoder.test.ts

```diff
@@ -0,0 +1,105 @@
+import * as koapModule from '@juullabs/koap'
+const koap = koapModule.com.juul.koap
+
+import * as kotlin from '@juullabs/koap/node_modules/kotlin'
+
+test('decodeTcp :: TCP payload :: correctly decoded', () => {
+  
+    const encodedMessage = new Uint8Array([0xD1, 0x04, 0x02, 0x42, 0xB4, 0x69, 0x6E, 0x66, 0x6F, 0x44, 0x62, 0x61, 0x74, 0x74, 0xFF, 0x42, 0x61, 0x74, 0x6D, 0x61, 0x6E])
+
+    const decodedMessage = koap.decodeTcp(encodedMessage)
+    
+    expect(decodedMessage.code).toEqual(koap.Message.Code.Method.POST)
+    expect(decodedMessage.token).toEqual(new kotlin.Long(66))
+    expect(decodedMessage.options.modCount).toEqual(2)
+    expect(decodedMessage.payload).toEqual(new Uint8Array([0x42, 0x61, 0x74, 0x6d, 0x61, 0x6e]))
+});
+
+test('decodeUdp :: UDP payload :: correctly decoded', () => {
+  
+    const encodedMessage = new Uint8Array([0x42, 0x01, 0xFE, 0xED, 0xCA, 0xFE, 0xB5, 0x2F, 0x74, 0x65, 0x73, 0x74, 0xFF, 0x48, 0x65, 0x6C, 0x6C, 0x6F, 0x20, 0x55, 0x44, 0x50, 0x21])
+
+    const decodedMessage = koap.decodeUdp(encodedMessage)
+    
+    expect(decodedMessage.type).toEqual(koap.Message.Udp.Type.Confirmable)
+    expect(decodedMessage.code).toEqual(koap.Message.Code.Method.GET)
+    expect(decodedMessage.id).toEqual(0xFEED)
+    expect(decodedMessage.token).toEqual(new kotlin.Long(0xCAFE))
+    expect(decodedMessage.options.modCount).toEqual(1)
+    expect(decodedMessage.payload).toEqual(new Uint8Array([0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x55, 0x44, 0x50, 0x21]))
+});
+
+test('decodeWithUdpHeader :: UDP payload :: returns header correctly', () => {
+
+    const encodedMessage = new Uint8Array([0x42, 0x01, 0xFE, 0xED, 0xCA, 0xFE, 0xB5, 0x2F, 0x74, 0x65, 0x73, 0x74, 0xFF, 0x48, 0x65, 0x6C, 0x6C, 0x6F, 0x20, 0x55, 0x44, 0x50, 0x21])
+
+    const mType = koap.Message.Udp.Type.Acknowledgement
```

Comment:
This looks like Hungarian notation, which is an [abandoned style for Android code](https://jakewharton.com/just-say-no-to-hungarian-notation/); so could be slightly confusing for long time Android devs.

```suggestion
    const messageType = koap.Message.Udp.Type.Acknowledgement
```

or

```suggestion
    const type = koap.Message.Udp.Type.Acknowledgement
```